src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -348,6 +386,59 @@ private void EventLoop()
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ScheduleToProcessEvents()
+        {
+            // Schedule a thread pool work item to process events. Only one work item is scheduled at any given time to avoid
+            // over-parallelization. When the work item begins running, this field is reset to 0, allowing for another work item
+            // to be scheduled for parallelizing processing of events.
+            if (Interlocked.CompareExchange(ref _eventQueueProcessingRequested, 1, 0) == 0)
+            {
+                ThreadPool.UnsafeQueueUserWorkItem(this, preferLocal: false);
+            }
+        }
+
+        void IThreadPoolWorkItem.Execute()
+        {
+            // Indicate that a work item is no longer scheduled to process events. The change needs to be visible to enqueuer
+            // threads (only for EventLoop() currently) before an event is attempted to be dequeued. In particular, if an
+            // enqueuer queues an event and does not schedule a work item because it is already scheduled, and this thread is
+            // the last thread processing events, it must see the event queued by the enqueuer.
+            Interlocked.Exchange(ref _eventQueueProcessingRequested, 0);
+
+            ConcurrentQueue<SocketIOEvent> eventQueue = _eventQueue;
+            int startTimeMs = Environment.TickCount;
```

Comment:
nit: maybe move `startTimeMs` assignment below `TryDequeue`.

Reply:
Yea it would avoid the tick count when the queue is empty, will do